lib/database-ops.js

```diff
@@ -101,46 +101,79 @@ var changeUserPasswordInDatabase = function(userID, hashedPassword) {
 
 var startDatabaseClient = function(callback, options) {
     var dbURL = (options !== undefined && options.dbURL !== undefined) ? options.dbURL : url;
-    MongoClient.connect(dbURL, function(err, db) {
-        if (err) {
-            callback(err);
-            return;
-        }
+    MongoClient.connect(dbURL).then((db) => {
         logger.info("Successfully connected to database.");
         database = db;
 
-        var finishInitialization = function () {
-            imageEntryCollection = database.collection("image-entries");
+        var finishInitialization = function () {  
             commentsCollection = database.collection("comments");
             logCollection = database.collection("logs");
             actionHistoryCollection = database.collection("action-history");
 
             usersCollection.createIndex({ username: 1 }, { unique: true, sparse: true });
-            callback(err);
+            callback();
         };
 
-        database.listCollections().toArray(function(err, collections) {
-            if (err) {
-                throw err;
+        var createIndexForImageEntriesCollection = function(collection) {
+            collection.createIndex({ uploadeddate: 1 }, { 
```

Comment:
Would be nice to have a log statement for when the index is created. I think it can go into the callback of `createIndex` or from its promise.